cordapp-cpk/src/main/kotlin/net/corda/plugins/cpb/CpbTask.kt

```diff
@@ -0,0 +1,76 @@
+package net.corda.plugins.cpb
+
+import net.corda.plugins.cpk.CORDA_CPK_TYPE
+import net.corda.plugins.cpk.CPK_FILE_EXTENSION
+import net.corda.plugins.cpk.SignJar.Companion.sign
+import net.corda.plugins.cpk.Signing
+import org.gradle.api.file.DuplicatesStrategy
+import org.gradle.api.tasks.AbstractCopyTask
+import org.gradle.api.tasks.bundling.Jar
+import org.gradle.api.tasks.bundling.ZipEntryCompression
+import java.nio.file.Files
+import java.util.jar.JarInputStream
+import java.util.zip.ZipInputStream
+
+open class CpbTask : Jar() {
+
+    companion object {
+        private const val CPB_ARTIFACT_CLASSIFIER = "package"
+        const val CPB_FILE_EXTENSION = "cpb"
+        private const val CPK_FILE_SUFFIX = ".$CPK_FILE_EXTENSION"
+        private const val JAR_FILE_SUFFIX = ".jar"
+        private val EXCLUDED_CPK_TYPES = setOf("corda-api")
+    }
+
+    val signing : Signing = project.objects.newInstance(Signing::class.java)
+
+    init {
+        archiveClassifier.set(CPB_ARTIFACT_CLASSIFIER)
+        archiveExtension.set(CPB_FILE_EXTENSION)
+        dirMode = Integer.parseInt("555", 8)
+        duplicatesStrategy = DuplicatesStrategy.FAIL
+        fileMode = Integer.parseInt("444", 8)
+        entryCompression = ZipEntryCompression.STORED
+        manifestContentCharset = "UTF-8"
+        metadataCharset = "UTF-8"
+        includeEmptyDirs = false
+        isCaseSensitive = true
+        isPreserveFileTimestamps = false
+        isReproducibleFileOrder = true
+        isZip64 = true
+
+        doLast {
+            if(signing.enabled.get()) {
+                sign(signing, archiveFile.get().asFile)
+            }
+        }
```

Comment:
It's probably better to do this from the plugin, because the `signing` block you want to use belongs to the `cordapp` extension and not this task.